File: src/shared/api/agentConfigs.ts
import { invoke } from "@tauri-apps/api/core";

// Types

export interface AgentConfig {
  id: string;
  name: string;
  description?: string;
  instructions: string;
  filePath: string;
  source: string;
  lastModified: string;
}

export interface CreateAgentConfigRequest {
  name: string;
  description?: string;
  instructions: string;
}

export interface UpdateAgentConfigRequest {
  name?: string;
  description?: string;
  instructions?: string;
}

export async function listAgentConfigs(): Promise<AgentConfig[]> {
  return invoke("list_agent_configs");
}

export async function getAgentConfig(id: string): Promise<AgentConfig> {
  return invoke("get_agent_config", { id });
}

export async function createAgentConfig(
  request: CreateAgentConfigRequest,
): Promise<AgentConfig> {
  return invoke("create_agent_config", { request });
}

export async function updateAgentConfig(
  id: string,
  request: UpdateAgentConfigRequest,
): Promise<AgentConfig> {
  return invoke("update_agent_config", { id, request });
}

export async function deleteAgentConfig(id: string): Promise<void> {
  return invoke("delete_agent_config", { id });
}

export async function refreshAgentConfigs(): Promise<AgentConfig[]> {
  return invoke("refresh_agent_configs");
}

File: src-tauri/src/commands/agent_configs.rs
use crate::services::agent_configs::AgentConfigStore;
use crate::types::agents::{AgentConfig, CreateAgentConfigRequest, UpdateAgentConfigRequest};
use tauri::State;

#[tauri::command]
pub fn list_agent_configs(store: State<'_, AgentConfigStore>) -> Vec<AgentConfig> {
    store.list()
}

#[tauri::command]
pub fn get_agent_config(
    store: State<'_, AgentConfigStore>,
    id: String,
) -> Result<AgentConfig, String> {
    store
        .get(&id)
        .ok_or_else(|| format!("Agent config '{}' not found", id))
}

#[tauri::command]
pub fn create_agent_config(
    store: State<'_, AgentConfigStore>,
    request: CreateAgentConfigRequest,
) -> Result<AgentConfig, String> {
    store.create(request)
}

#[tauri::command]
pub fn update_agent_config(
    store: State<'_, AgentConfigStore>,
    id: String,
    request: UpdateAgentConfigRequest,
) -> Result<AgentConfig, String> {
    store.update(&id, request)
}

#[tauri::command]
pub fn delete_agent_config(
    store: State<'_, AgentConfigStore>,
    id: String,
) -> Result<(), String> {
    store.delete(&id)
}

#[tauri::command]
pub fn refresh_agent_configs(store: State<'_, AgentConfigStore>) -> Vec<AgentConfig> {
    store.refresh()
}
